app/src/main/java/org/kiwix/kiwixmobile/search/AutoCompleteAdapter.java

```diff
@@ -82,9 +83,17 @@ public String getItem(int index) {
     }
   }
 
-  @Override
-  public Filter getFilter() {
-    return mFilter;
+  @Override public int getItemCount() {
+    return mData.size();
+  }
+
+  public class ViewHolder extends RecyclerView.ViewHolder {
+    TextView txt;
+
+    public ViewHolder(@NonNull View itemView) {
+      super(itemView);
+      txt = itemView.findViewById(android.R.id.text1);
+    }
   }
 
   class KiwixFilter extends Filter {
```

Comment:
I really really hate this Filter. mostly because it doesn't actually filter a goddamn thing, it actually populates an entirely new dataset.
It would be very cool if the searchview sent queries to the presenter and the presenter were to respond with new data for the recyclerview to render This will get rid of your "invalidated" problem